cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -202,40 +206,70 @@ countConsumed = do
     pure countRows
   pure $ maybe 0 unValue (listToMaybe res)
 
+deleteAndUpdateConsumedTxOut :: MonadIO m => Trace IO Text -> Word64 -> ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut trce blockNoDiff = do
+  maxTxInId <- findMaxTxInId blockNoDiff
+  case maxTxInId of
+    Left errMsg -> liftIO $ logInfo trce $ "No tx_out were deleted: " <> errMsg
+    Right mxtxid -> do
+      migrateNextPage mxtxid False 0
+  where
+    migrateNextPage :: MonadIO m => TxInId -> Bool -> Word64 -> ReaderT SqlBackend m ()
+    migrateNextPage mxTxInId ranCreateConsumed offst = do
+      page <- getInputPage offst pageSize
+      res <- foldM (handlePageEntry mxTxInId) ranCreateConsumed page
+      when (fromIntegral (length page) == pageSize) $
+        migrateNextPage mxTxInId res $!
+          offst
+            + pageSize
+
+    handlePageEntry :: MonadIO m => TxInId -> Bool -> (TxInId, TxId, Word64) -> ReaderT SqlBackend m Bool
+    handlePageEntry maxTxInId rcc (txInId, txId, index)
+      | txInId <= maxTxInId = do
+          deleteTxOutConsumed txId index
```

Comment:
It may be worth examining if a whole page can be deleted once, with a single db query, instead of doing it entry by entry. This optimization can be done on a different pr.